hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/inject/AvroUpsertTarget.java

```diff
@@ -146,4 +166,9 @@ public Object conclude() {
         this.record = null;
         return record;
     }
+
+    @FunctionalInterface
+    private interface Injector {
```

Comment:
this interface is rather similar to `UpsertTarget`. UpsertTarget works on current object. Might be possible to create `NestedAvroUpsertTarget`. Note that `RECORD` handling does similar steps as init/conclude. Such design would more symmetric but I am not sure if simpler.

Reply:
In #25349, I allowed writing to top-level fields and this asymmetry became more visible. I will fix this issue in that PR.